regulation-worker/internal/delete/api/api.go

```diff
@@ -39,6 +41,31 @@ func (api *APIManager) Delete(ctx context.Context, job model.Job, destConfig map
 
 	bodySchema := mapJobToPayload(job, strings.ToLower(destName), destConfig)
 	pkgLogger.Debugf("payload: %#v", bodySchema)
+	pkgLogger.Debugf("Regulation Delete API called:\n")
+
+	var tokenStatusCode int
+	var accountSecretInfo *oauth.AuthResponse
+	// identifier to know if the destination supports OAuth
+	// TODO: "rudderAccountId" has to change to "rudderUserDeleteAccountId"
+	rudderUserDeleteAccountId, delAccountIdExists := destConfig["rudderAccountId"]
```

Comment:
Isn't the presence of a valid value in `rudderUserDeleteAccountId` enough to determine that a destination supports user deletion via OAuth?

Reply:
The thing is we can do this check through `DestinationDefinition.Config.auth.type` check rather than `Destination.Config`. But we need to decide if going via DestinationDefinition makes sense over Destination.Config or vice-versa

Reply:
Ok. Keeping this open as a reminder to address it.